### PR TITLE
Add support for Smooch.io

### PR DESCRIPTION
**Pull request by gozman**
Ben & team,

As discussed, here's my PR for connecting BotKit to Smooch.
Let's get this working!

Mike

### CODE REVIEW

File: lib/SmoochBot.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    _______..___  ___.   ______     ______     ______  __    __  .______     ______   .___________.
+    /       ||   \/   |  /  __  \   /  __  \   /      ||  |  |  | |   _  \   /  __  \  |           |
+   |   (----`|  \  /  | |  |  |  | |  |  |  | |  ,----'|  |__|  | |  |_)  | |  |  |  | `---|  |----`
+    \   \    |  |\/|  | |  |  |  | |  |  |  | |  |     |   __   | |   _  <  |  |  |  |     |  |     
+.----)   |   |  |  |  | |  `--'  | |  `--'  | |  `----.|  |  |  | |  |_)  | |  `--'  |     |  |     
+|_______/    |__|  |__|  \______/   \______/   \______||__|  |__| |______/   \______/      |__|     
+                                                                                                    
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
```
Comment (anonrig) [inline lib/SmoochBot.js]: Please don't do this. We're using JSDoc.

File: lib/SmoochBot.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    return smooch_botkit;
+
+  }
+
+  // set up a web route that is a landing page
+  smooch_botkit.createHomepageEndpoint = function(webserver) {
+
+    smooch_botkit.log('** Serving app landing page at : http://MY_HOST:' + smooch_botkit.config.port + '/');
+
+    // FIX THIS!!!
```
Comment (anonrig) [inline lib/SmoochBot.js]: Please don't submit buggy code to production env.

File: lib/SmoochBot_worker.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  }
+
+  bot.startConversation = function(message, cb) {
+    botkit.startConversation(this, message, cb);
+  };
+
+  /***
+    This handles the particulars of finding an existing conversation or
+    topic to fit the message into...
+   ***/
```
Comment (anonrig) [inline lib/SmoochBot_worker.js]: Please review JSDoc.

File: package.json [hunk 4]
```diff
@@ -4,11 +4,13 @@
   "description": "Building blocks for Building Bots",
   "main": "lib/Botkit.js",
   "dependencies": {
+    "babel-runtime": "^6.3.19",
```
Comment (anonrig) [inline package.json]: Why did you need `babel`?

**Comment by RafaelCosman** [comment]
Hey @gozman, thanks for submitting this PR. Can you address @anonrig's comments so we can get this merged? Thanks!
